File: f#/words/test/NumericalTest.fs
namespace test.NumericalTests

open NUnit.Framework
open lib.Numerical

[<TestFixture>]
module NumericalTests =

    [<Test>]
    let ``Fibonacci`` () =
        Assert.AreEqual(-1, Numerical.fibonacci -3)
        Assert.AreEqual(55, Numerical.fibonacci 10)
        Assert.AreEqual(610, Numerical.fibonacci 15)

    [<Test>]
    let ``Factorial`` () =
        Assert.AreEqual(-1, Numerical.factorial -3)
        Assert.AreEqual(120, Numerical.factorial 5)
        Assert.AreEqual(3628800, Numerical.factorial 10)

    [<Test>]
    let ``Greatest Common Divisor`` () =
        Assert.AreEqual(6, Numerical.gcd 48 18)
        Assert.AreEqual(25, Numerical.gcd 100 25)
        Assert.AreEqual(1, Numerical.gcd 17 13)

    [<Test>]
    let ``Lowest Common Multiple`` () =
        Assert.AreEqual(12, Numerical.lcm 4 6)
        Assert.AreEqual(42, Numerical.lcm 21 6)
        Assert.AreEqual(21, Numerical.lcm 7 3)

File: _experimental/_zig/words/test/numerical_test.zig
const std = @import("std");
const numerical = @import("numerical");

test "Fibonacci" {
    try std.testing.expect(numerical.fibonacci(-3) == -1);
    try std.testing.expect(numerical.fibonacci(10) == 55);
    try std.testing.expect(numerical.fibonacci(15) == 610);
}

test "Factorial" {
    try std.testing.expect(numerical.factorial(-3) == -1);
    try std.testing.expect(numerical.factorial(5) == 120);
    try std.testing.expect(numerical.factorial(10) == 3628800);
}

test "Greatest Common Divisor" {
    try std.testing.expect(numerical.gcd(48, 18) == 6);
    try std.testing.expect(numerical.gcd(100, 25) == 25);
    try std.testing.expect(numerical.gcd(17, 13) == 1);
}

test "Lowest Common Multiple" {
    try std.testing.expect(numerical.lcm(4, 6) == 12);
    try std.testing.expect(numerical.lcm(21, 6) == 42);
    try std.testing.expect(numerical.lcm(7, 3) == 21);
}
